### ISSUE TITLE
Unable to install `rice-3.0.0` gem using Ruby 2.6.3 on macOS Monterey

### ISSUE THREAD

**Issue by ronaldtse** [metanorma/packed-mn#140]
On Ruby 2.6.3, the `rice-3.0.0` gem is broken with `arch` detection, and there is no way we can fix it without making a new gem here.

```
$ rbenv shell 2.6.3
$ irb
irb(main):001:0> RbConfig::CONFIG["arch"].split("-")[0]
=> ""
$ rbenv shell 3.0.3
$ irb
irb(main):001:0> RbConfig::CONFIG["arch"].split("-")[0]
=> "arm64"
```

https://github.com/jasonroelofs/rice/blob/9d0acb80d8654bfc6d0e65f35654d434ea66add9/extconf.rb#L48-L70
```ruby
  darwin_version = $1.to_i

  if darwin_version >= 10
    arch = RbConfig::CONFIG["arch"].split("-")[0]

    if arch == "universal"
      arch = `uname -m`.strip
    end

    other_opts = "--disable-dependency-tracking"
  else
    arch = `uname -p`.chomp
  end

  env = if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.0.0')
          "ARCHFLAGS='-arch #{arch}' CPPFLAGS='-fdeclspec -arch #{arch}'"
        else
          "ARCHFLAGS='-arch #{arch}' CPPFLAGS='-arch #{arch}'"
        end
end

system "sh bootstrap"
system "#{env} sh configure --with-ruby=#{with_ruby} --prefix=#{prefix_dir} #{other_opts}"
```

This particular code is used to detect `arch` and pass as `CPPFLAGS`, however, it fails because the result is an empty string (``).

```
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether the C++ compiler works... no
configure: error: in `/Users/me/.rbenv/versions/2.6.3/lib/ruby/gems/2.6.0/gems/rice-3.0.0':
configure: error: C++ compiler cannot create executables
See `config.log' for more details

Makefile not found
```

The config.log shows:
```
configure:3063: clang++ -qversion >&5
clang-13: error: unknown argument '-qversion'; did you mean '--version'?
clang-13: error: no input files
configure:3074: $? = 1
configure:3094: checking whether the C++ compiler works
configure:3116: clang++  -arch   conftest.cpp  >&5
clang-13: error: invalid arch name '-arch conftest.cpp'
clang-13: error: no input files
configure:3120: $? = 1
configure:3160: result: no
configure: failed program was:
| /* confdefs.h */
| #define PACKAGE_NAME "rice"
| #define PACKAGE_TARNAME "rice"
| #define PACKAGE_VERSION "1.1"
| #define PACKAGE_STRING "rice 1.1"
| #define PACKAGE_BUGREPORT ""
| #define PACKAGE_URL ""
...
```

The arch name is invalid because it was empty.

The only way out seems to be upgrading to rice-4.0.

**Comment by ronaldtse** [comment]
Ruby 2.7.3 works though.

**Comment by ronaldtse** [comment]
Fixed, we're using Rice 4.0.0 now.
